### DISCUSSION TITLE
Identifying and importing missing photos that were expected to be imported

### DISCUSSION THREAD

**Discussion by timbuktumor** [photoprism/photoprism#3499] [General]
I'm trying out Photoprism and having a heck of a time identifying missing photos from an import set.  I can't find easy way via UI to correlate back to the source file used during import.  Large imports seem to exceed log buffer (or I'm not looking correctly).  Only way I've been able to identify is through SQL query and compare the original file name with source folder file names.  After pinpointing the 3 missing files, I tried to import one of them and saw this show up in the log:
 
```
time="2023-06-22T19:47:59-07:00" level=info msg="media: 20210524_181406.jpg was taken at 2021-05-24 18:14:15 +0000 UTC (meta)"
time="2023-06-22T19:47:59-07:00" level=info msg="import: 2021/05/20210524_181415_C637AE12.jpg already exists"
time="2023-06-22T19:47:59-07:00" level=info msg="import: deleted 20210524_181406.jpg (already exists)"
```

Any explanation why it's reporting a different file exists?  I would expect it would complain 181406 would be a dupe and I can't find any existence of 181415.  Attempting to search using _filename:"2021/05/20210524_181415_C637AE12.jpg"_ returns no picture.  One thing in common with these 3 missing photos is they are panoramic. 

The UI does show this in errors, but it's not super helpful as 20210524_181415.jpg is not a valid "original" from my source folder.
`import: failed creating thumbnails for 2021/05/20210524_181415_C637AE12.jpg (EOF while decoding)`

Thanks!

**Comment by graciousgrey** [comment]
During [import](https://docs.photoprism.app/user-guide/library/import/)  files are moved from import to originals and get a canonical name. That's why the filenames differ in the log messages: I assume 20210524_181406.jpg is the original filename of the file in your import folder, after it has been copied/moved to originals it gets a new name: 2021/05/20210524_181415_C637AE12.jpg.

In your case the import worked. So the file should be physically present in the originals folder with the following path: 2021/05/20210524_181415_C637AE12.jpg.

However as indicated by the error message the file seems to be broken (has invalid metadata) and cannot be indexed. This is is why it does not appear in the UI when searching for the filename. 

Are your files from a Samsung Phone? Then this might be of interest to you:

- https://github.com/photoprism/photoprism/issues/3363
- https://github.com/photoprism/photoprism/issues/1407#issuecomment-1532767800

**Comment by timbuktumor** [comment]
> Are your files from a Samsung Phone?

Thanks for your reply.  Yes, they are from Samsung Galaxy S9.  It's possible the "import: failed creating thumbnails" errors are related to this, however I'm confused why Photoprism thinks these are duplicates.

> I assume 20210524_181406.jpg is the original filename of the file in your import folder, after it has been copied/moved to originals it gets a new name: 2021/05/20210524_181415_C637AE12.jpg

Yes, 20210524_181406.jpg is the original filename.  If you notice, the new canonical name contains 181415 instead of 181406.  Every other successful import has a canonical name that contains the original filename.  This one slightly changes for unknown reason.  I would expect it to be 2021/05/20210524_1814**06**_C637AE12.jpg

> In your case the import worked. So the file should be physically present in the originals folder with the following path: 2021/05/20210524_181415_C637AE12.jpg.

That's the rub.  It doesn't look like it imported because apparently it already exists, however I cannot find the supposed duplicate anywhere.  Neither 2021/05/20210524_181406_C637AE12.jpg or 2021/05/20210524_181415_C637AE12.jpg can be found.

**Comment by timbuktumor** [comment]
> Every other successful import has a canonical name that contains the original filename.

After comparing database records, I guess this isn't 100% accurate.  At the end of the day, my point is the canonical name used in an import error message doesn't help determine which file on my computer triggered it.

**Reply by lastzero** [reply]
The original name is stored in the files table though.

**Comment by timbuktumor** [comment]
> The original name is stored in the files table though.

Not in this case as the files I'm looking for did not import.  For example:

```
MariaDB [photoprism]> select * from errors where id=2;
+----+---------------------+-------------+--------------------------------------------------------------------------------------------------+
| id | error_time          | error_level | error_message                                                                                    |
+----+---------------------+-------------+--------------------------------------------------------------------------------------------------+
|  2 | 2023-06-22 21:52:36 | error       | import: failed creating thumbnails for 2021/05/20210524_181415_C637AE12.jpg (EOF while decoding) |
+----+---------------------+-------------+--------------------------------------------------------------------------------------------------+
1 row in set (0.001 sec)

MariaDB [photoprism]> select original_name, file_name from files where file_name = '2021/05/20210524_181415_C637AE12.jpg';
Empty set (0.000 sec)

MariaDB [photoprism]> 
```

**Comment by graciousgrey** [comment]
> Neither 2021/05/20210524_181406_C637AE12.jpg or 2021/05/20210524_181415_C637AE12.jpg can be found.

Do you mean using the Web UI or directly on the filesystem within the originals folder? 

I agree it might be helpful to display the original name next to the canonical name in error messages in case files have been imported. But I am not sure if that impacts indexing performance. 

With our preview build broken samsung files will be repaired automatically so in the future this error should not occur again :)

**Comment by timbuktumor** [comment]
> With our preview build broken samsung files will be repaired automatically so in the future this error should not occur again :)

Very cool.  Amazing stuff!

> Do you mean using the Web UI or directly on the filesystem within the originals folder?

Using the search options in Web UI or direct database SQL query.

> I agree it might be helpful to display the original name next to the canonical name in error messages in case files have been imported.

Yes it would.  If you refer to 1st post, the first error clues me back to the original filename in filesystem, but this was seen during a single file import example.  It required spending time comparing the database filenames with the filesystem filenames to determine missing files before I could re-create the error for this post.  In other words, that error was not visible during a larger import of 2000+ files and when import was complete, the only visible error in UI was the 2nd error referring to a canonical name which doesn't exist in the database.

Is there a logfile somewhere that contains 100% of the import history?  I was looking at console output which is buffered.  Some method, (logfile or new table) to review these actions could be helpful if it doesn't exist already.

**Reply by graciousgrey** [reply]
You can view the logs as described here:

  https://docs.photoprism.app/getting-started/troubleshooting/docker/#viewing-logs

In addition errors and warnings are saved in the database.

I also created an issue for more helpful error messages, feel free to leave a :+1: in there :)
- https://github.com/photoprism/photoprism/issues/3556
